include/revng/PipelineC/Prototypes.h

```diff
@@ -458,6 +470,21 @@ rp_container_extract_one(rp_container *container, rp_target *target);
 
 /** \} */
 
+/**
+ *
+ * \defgroup rp_diff_map rp_diff_map methods
+ * \{
+ */
+void rp_diff_map_destroy(rp_diff_map *to_free);
+
+/**
+ * \returns nullptr if global_name did not named a global variable in the
+ * diff_map else return the serialized diff of the indicated global
+ */
+const char *rp_diff_map_create_diff(rp_diff_map *map, const char *global_name);
```

Comment:
I'm confused. Is this WIP?

Reply:
why? it follows the convetions that create means a owning pointer. it returns the serialized content of the map

Reply:
Ahh, so this is `rp_diff_map_serialize`. The idea of always having `create_` in the name is not working very well.
It makes certain names very counter-intuitive and we're not being consistent (`rp_manager_run_analysis` returns an owning pointer, but it does not contain `create`).

Shall we try to change approach?
What about:
```c++
#define OWNING(x) x

/// \note The returned pointer is owning and needs to be freed by the caller
OWNING(const char *) rp_diff_map_serialize(...);
```

Reply:
yeah i guess it is more reasonable